packages/react-date-time/src/components/DatePicker/DatePicker.tsx

```diff
@@ -0,0 +1,253 @@
+import * as React from 'react';
+import flatpickr from 'flatpickr';
+import { css } from '@patternfly/react-styles';
+import FlatpickrLanguages from 'flatpickr/dist/l10n';
+import 'flatpickr/dist/themes/light.css';
+import '@patternfly/patternfly/patternfly-date-picker.css';
+import styles from '@patternfly/react-styles/css/components/DatePicker/date-picker';
+import { DatePickerLocales } from './DatePickerUtils';
+import { TextInput } from '@patternfly/react-core/dist/js/components/TextInput/TextInput';
+
+export interface DatePickerProps
+  extends Omit<React.HTMLProps<HTMLInputElement>, 'onChange' | 'onFocus' | 'onBlur' | 'disabled' | 'ref'> {
+  /** Additional classes added to the date time picker. */
+  className?: string;
+  /** Flag indicating whether the value of the date picker is invalid. */
+  invalid?: boolean;
+  /** Accessible label for the date picker */
+  'aria-label'?: string;
+  /** The minimum date that a user can start picking from. */
+  minDate?: string;
+  /** The maximum date that a user can pick to. */
+  maxDate?: string;
+  /** The case sensitive pattern of tokens representing desired readable format of the date. 'MM/DD/YYYY' by default. */
+  dateFormat?: string;
+  /** Flag indicating the date picker is disabled*/
+  isDisabled?: boolean;
+  /** Specify the locale of the date. */
+  locale?: DatePickerLocales;
+  /** String to display in the empty date picker field as a hint for the expected date format */
+  placeholder?: string;
+  /** A date string. The format could be the expected date format, or an ISO 8601 formatted date string */
+  value?: string;
+  /** Error message to display when the date is provided in an invalid format. */
+  invalidFormatErrorMessage?: string;
+  /** Error message to display when the date is provided in outside the valid date range. */
+  dateOutOfRangeErrorMessage?: string;
+  /** Error message to display when the date is provided before the provided minDate. */
+  beforeMinDateErrorMessage?: string;
+  /** Error message to display when the date is provided after the provided maxDate. */
+  afterEndDateErrorMessage?: string;
+}
+
+interface DatePickerState {
+  invalid: boolean;
+  invalidText: string;
+}
+
+function toNumber(str: string) {
+  if (str) {
+    return Number(str);
+  }
+  return NaN;
+}
+// Supports (yyyy|mm|dd) to parse string to Date object
+function parseYYYYMMDD(date: string, format: string) {
```

Comment:
flatpickr has a `parseDate` function, i don't think we need these functions here